app/controllers/application_controller.rb

```diff
@@ -137,4 +129,12 @@ def set_locale
   def sp_session
     session.fetch(:sp, {})
   end
+
+  def create_branded_experience
+    return unless session[:sp]
+
+    @sp_logo = current_sp.logo
```

Comment:
this might be for a separate PR, but I wonder if it would be better to have an `@sp` or `@current_sp` instance var instead of all 3 of these, but to have this info available from that instance var. In general, having more than a couple of instance vars available in any one view is a 🔴  flag to me.